src/hooks/useArNSContract/useArNSContract.tsx

```diff
@@ -0,0 +1,47 @@
+import { LocalFileSystemDataProvider } from '../../services/arweave/LocalFilesystemDataProvider';
+import { useStateValue } from '../../state/state';
+import { useState, useEffect } from 'react';
+
+const localContractAddress = 'bLAgYxAdX2Ry-nt6aH2ixgvJXbpsEYm28NgJgyqfs-U';
+
+export default function useArNSContract() {
+  const [{ arnsSourceContract }, dispatch] = useStateValue();
+  const [sendingContractState, setSendingContractState] = useState(false);
+
+  try {
```

Comment:
let's move the try into `dispatchNewContractState` as well as `localProvider`

so use effect will be at the top level, and try/catch will be IN `dispatchNewContractState`